### ISSUE TITLE
PLIP: allow custom caption in relation behavior

### ISSUE THREAD

**Issue by mtrebron** [plone/Products.CMFPlone#2979]
## PLIP (Plone Improvement Proposal)

When a plone.app.relation field is added as a behavior to a content type, it will display with a fixed caption "Related content" above the items. It would be useful for a content manager to be able to customize this text, depending on the context in which it is presented e.g. "Favorite authors" or "Suggested articles"

This can be achieved by adding a second field to the behavior which may be used to optionally override the default text. The proposed change also includes updating the plone.belowcontentbody.relateditems viewlet provided by plone.app.layout

Overall, this change would improve consistency between similar behaviors (e.g. the ILeadImageBehavior in plone.app.contenttypes) 

## Abstract

Add a customizable caption field to relations

## Motivation

In our use case, relations can be displayed outside of their direct context (e.g. in a mega-menu). By allowing the caption to be customized, a better indication can be given of what is being presented.  

## Assumptions

The caption field would be optional, and existing relations which do not use the field mantain their current functionality.

## Proposal & Implementation

I have cloned both affected packages and am working on an implementation for Plone 5.2

## Deliverables

New text field, updated view.

## Risks

None

## Participants

@mtrebron

**Comment by MrTango** [comment]
Sounds like a good idea to me.

**Comment by rodfersou** [comment]
+1

**Comment by tisto** [comment]
Move to stalled. No activity since 2022.

**Comment by rohnsha0** [comment]
@mtrebron are you still working on it? i would like to take over it... and push a PR

**Comment by davisagli** [comment]
I don't agree with this PLIP as it is written. It's already possible to create a custom behavior with a relation field that has its own title. Adding a new field to specify the label shown to anonymous users sounds like it could be confusing for editors and would lead to inconsistent labels since they would need to add it to each content item.

In general different fields should be used to store different things, instead of trying to use one field to store different things.

**Comment by rohnsha0** [comment]
![edit](https://github.com/user-attachments/assets/ecec1d96-d102-4bd6-8f31-8336d5a52a20)

![view](https://github.com/user-attachments/assets/ec48ce7d-2465-4091-a004-dbe5e0d5bcf8)

@davisagli I somewhat implemented a new field in https://github.com/plone/plone.app.relationfield/blob/9af9389fc422208166f4a79f2fc6e535477ec896/plone/app/relationfield/behavior.py#L16

and later implemented the same in p.a.layout... doesnt that makes sense?

**Comment by mtrebron** [comment]
@davisagli I see the problem now. The [ILeadImageBehavior](https://github.com/plone/plone.app.contenttypes/blob/master/plone/app/contenttypes/behaviors/leadimage.py) in plone.app.contenttypes is optional. 

FWIW we are not actively using this feature anymore.

**Comment by yurj** [comment]
Can we close this PLIP?

**Comment by ale-rt** [comment]
Recently I created a package to experiment with something similar, see https://github.com/collective/collective.z3cform.captionedrelationfield/.
In particular check https://github.com/collective/collective.z3cform.captionedrelationfield/blob/04ea31314b62f2bea7821e3ed5cb94c131d5e141/src/collective/z3cform/captionedrelationfield/tests/behaviors/__init__.py

![Image](https://github.com/user-attachments/assets/a151fe7c-c38a-45ef-8ad7-d4da6348afc3)

You need latest coredev to make it work plus this patch https://github.com/plone/plone.z3cform/pull/30
